File: 1287-element-appearing-more-than-25-in-sorted-array/1287-element-appearing-more-than-25-in-sorted-array.py
class Solution:
    def findSpecialInteger(self, arr: List[int]) -> int:
        res = arr[0]
        count = 1
        percentage = len(arr)//4
        
        for num in arr[1:]:
            if num == res:
                count += 1
            else:
                res = num
                count = 1
            
            if count > percentage:
                return num
        
        return res

File: 1287-element-appearing-more-than-25-in-sorted-array/1287-element-appearing-more-than-25-in-sorted-array.js
/**
 * @param {number[]} arr
 * @return {number}
 */
var findSpecialInteger = function(arr) {
    let currNum = arr[0];
    let count = 1;
    const arrLength = arr.length;
    const threshold = Math.floor(arrLength/4);
    
    for (let i = 1; i < arrLength; i++){
        if (arr[i] == currNum){
            count += 1;
        }
        else {
            currNum = arr[i];
            count = 1;
        }
        
        if (count > threshold){
            return arr[i];
        }
    }
    
    return currNum;
};